src/alfasim_sdk/result_reader/aggregator.py

```diff
@@ -1433,3 +1444,87 @@ def merge_metadata(a_metadata, b_metadata, *, source, update=(), min_=(), max_=(
         result_directory=r_a.result_directory,
         app_version_info=app_version_info,
     )
+
+
+def open_global_sensitivity_analysis_result_file(results_directory: Path) -> h5py.File:
+    """
+    Open a global sensitivity analysis result file.
+    """
+
+    h5py_file = h5py.File
+    if h5py.version.version_tuple[:2] >= (3, 5):
+        h5py_file = functools.partial(h5py.File, locking=RESULT_FILE_LOCKING_MODE)
+    filename = results_directory / "uq_result"
+    return _open_result_file(filename, h5py_file)
+
+
+def get_empty_global_sensitivity_analysis_result_metadata(
+    result_directory: Path,
+) -> UQResultMetadata:
+    """
+    Obtain an empty global sensitivity analysis meta-data.
+    """
+
+    return UQResultMetadata(
+        global_gsa_coefficients={}, result_directory=result_directory
+    )
+
+
+def read_global_sensitivity_analysis_meta_data(
+    result_directory: Path,
+) -> UQResultMetadata:
+    """
+    Read the global sensitivity analysis metadata persisted in a result file.
+    """
+
+    if not result_directory.is_dir():
+        return get_empty_global_sensitivity_analysis_result_metadata(
+            result_directory=result_directory,
+        )
+
+    if not Path(result_directory / "uq_result").is_file():
+        return get_empty_global_sensitivity_analysis_result_metadata(
+            result_directory=result_directory,
+        )
```

Comment:
This is questionable... if a file does not exist, it will return an empty metadata instead. We  don't see that usually, usually we get an error instead. Why did you decide on this design?

If it was not for a very good reason, we should probably behave like everybody else, and let the caller check if the file exists or not in order to decide what to do.

Reply:
Raising will require the generator source to handle that just to behave as having received an empty metadata.

Reply:
A fair trade for a "sane" API IMHO.